src/bits.hpp

```diff
@@ -27,61 +27,63 @@
 // 64 * 2^16. 2^17 values, each value can store 64 bits.
 #define kMaxSizeBits 8388608
 
-// A stack vector of length 5, having the functions of std::vector needed for Bits.
-struct SmallVector {
-    typedef uint16_t size_type;
+template <class item_type, class size_type_arg, unsigned capacity>
+class InlineVector {
+  public:
+    typedef size_type_arg size_type;
 
-    SmallVector() noexcept { count_ = 0; }
+    InlineVector() noexcept { count_ = 0; }
 
-    uint64_t& operator[](const uint16_t index) { return v_[index]; }
+    item_type& operator[](const size_type index) {
+        assert(index < capacity);
+        return v_[index];
+    }
 
-    uint64_t operator[](const uint16_t index) const { return v_[index]; }
+    item_type operator[](const size_type index) const {
+        assert(index < capacity);
+        return v_[index];
+    }
 
-    void push_back(uint64_t value) { v_[count_++] = value; }
+    void push_back(item_type value) {
+        assert(count_ < capacity);
+        v_[count_++] = value;
+    }
 
-    SmallVector& operator=(const SmallVector& other)
-    {
+    InlineVector& operator=(const InlineVector& other) {
```

Comment:
while you're at it, this function should probably be ref-qualified.
```suggestion
    InlineVector& operator=(const InlineVector& other) & {
```